src/translate.py

```diff
@@ -2623,9 +2649,9 @@ def handle_add(args: InstrArgs) -> Expression:
     rhs = args.reg(2)
     stack_info = args.stack_info
     type = Type.intptr()
-    if lhs.type.is_pointer():
+    if lhs.type.is_pointer_or_array():
         type = Type.ptr()
-    elif rhs.type.is_pointer():
+    elif rhs.type.is_pointer_or_array():
```

Comment:
hm, can array actually happen here? I feel like all expressions in registers should have decayed types. not quite assert-worthy though, and I guess it's fine to handle it like pointers

Reply:
Yeah I think you're right; it's in a register, an array shouldn't be possible.
I'll leave a comment about this, in case it comes up in a later refactor? (I think leaving it as `.is_pointer_or_array()` will also help later refactors, even if it's not needed?)

Reply:
Some kind of assert-only-in-debug-mode would be useful here. If it comes up we'd want to know about it since it's a probable bug, but at the same time it's a complex invariant so it's hard to be 100% sure about a hard assert, and there's an easy fallback path. Maybe too much effort to be worth introducing machinery for, though.